### ISSUE TITLE
Cannot add item in List using pnp in Sharepoint hosted add-in(SPOnline) , 

### ISSUE THREAD

**Issue by sbhamouda** [SharePoint/PnP-JS-Core#583]
[x] Question 

Hi all 
I'm developing a SharePoint hosted add-in (web addin) and am stuck with this error for a while I don't know am doing things wrong or is it a limitation of the crossdomain pnp ?
I'm trying to add an item in my list which hosted on the SharePoint site from a SharePoint web add-in 

so i added the pnp /promise / fetch script as bellow: in my pages and also my js file "addDero"
<script type="text/javascript" src="../Scripts/fetch.js"></script>
    <script type="text/javascript" src="../Scripts/es6-promise.js"></script>
    <script type="text/javascript" src="../Scripts/pnp.js"></script>
    <script type="text/javascript" src="../Scripts/addDero.js"></script> 

in my js file addDero ,i am running this script when filling a form :
'use strict';
function pnpReady() {
    var addinweb = getUrlParamByName("SPAppWebUrl");
    var hostweb = getUrlParamByName("SPHostUrl");
    $("#SendDeroBtn").click(function () {
        CreateItem(addinweb, hostweb)     
    });

function CreateItem(addinweb,hostweb) {
var d = $.Deferred();
    
    $pnp.sp.crossDomainWeb(addinweb, hostweb).lists.getByTitle('Derogation').items.add({
        Title: 'newPnpitem',
        Description: 'this item was added by pnp from sharepoint hosted add-in'
    }).then(function (response) {
        d.resolve(response);
    });
}

function getUrlParamByName(name) {
    name = name.replace(/[\[]/, "\\[").replace(/[\]]/, "\\]");
    var regex = new RegExp("[\\?&]" + name + "=([^&#]*)");
    var results = regex.exec(location.search);
    return results === null ? "" : decodeURIComponent(results[1].replace(/\+/g, " "));
}

Can any one help me out with this issue and give some help to solve it am new with pnp js core some am not sure that am doing right? any help would be appreciated

**Comment by patrick-rodgers** [comment]
What is the error you are getting? Sorry but your example is hard to follow so unsure where exactly I should be looking.

**Comment by simkessy** [comment]
Pnp returns a promise so you don't need to create your own using jquery. Do this instead: 

```
function CreateItem(addinweb,hostweb) {
  return $pnp.sp.crossDomainWeb(addinweb, hostweb).lists.getByTitle('Derogation').items.add({
    Title: 'newPnpitem',
    Description: 'this item was added by pnp from sharepoint hosted add-in'
  })
  .then(function (response) {
    return response
  })
  .catch(function (error) {
    return error
  });
}
```

**Comment by sbhamouda** [comment]
@patrick-rodgers  i get some time 400 error when i want to use lookup field value in the item that i want to add 

am get the the lookup field Id from a drop down so the fist part is like this : 

in the first step i need to get the lookup field value based on selected  drop down

UsageSelected= = $('#idUsageDropdown').val();
 var UsageId= $pnp.sp.crossDomainWeb(addinweb, hostweb).lists.getByTitle("USAGE").items.get().then(function (ListUsg) {
        for (var item in ListUsg) {
            if (ListUsg[itemvh].Title == UsageSelected) {
                return ListUsg[item].Id;
                alert("item found");
                console.log("ListUsg", ListUsg[item].Id);
            }
        }
       
    });

2/ then in the second step i need to send the item to list from the app 

-UsageVehiculeId  is the the id for the lookup field so i need to set it 
$pnp.sp.crossDomainWeb(addinweb, hostWeb).lists.getByTitle("Derogation").items.add({
        Title: "newTitle",
        UsageVehiculeId: UsageId,
    }).then(function (result) {
        alert("Item Added");
        //"result" will return couple of properties of newly created item. We can make use of them here
        console.log(result.data);
        console.log(result.data.Id);
    }).catch(function (err) {
        alert(err);
    });


NB: those 2 pnp call are in the same function that get triggered buy a submit btn . 

i didnt succeed and i get UsageVehiculeId="null" can any one give some help to resolve this issue and get my item added with the right lookup in it .

**Comment by patrick-rodgers** [comment]
If the UsageId value is null that means you aren't getting anything from the `$('#idUsageDropdown').val();` or at least not what you are expecting.

You can [see an example](https://github.com/SharePoint/PnP-JS-Core/wiki/Working-With:-Files#adding-files) of getting an inpute value and passing it to the library here. This deals with files, but the general idea is the same.

HTH

Going to close this as answered. If you require additional help specific to using this library please _reopen_ the ticket. For more general development questions I recommend [stackoverflow](https://stackoverflow.com/) or the [MS Tech community](https://techcommunity.microsoft.com/t5/SharePoint/ct-p/SharePoint) for MS related questions.
